### PR TITLE
First basic implementation using Grpc

### PR DESCRIPTION
**Pull request by bitmalloc**
Moving first java implementation code to it's own repo.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph4j/1)

### CODE REVIEW

File: src/main/java/io/dgraph/client/GrpcDgraphResult.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+public class GrpcDgraphResult extends DgraphResult<Graphresponse.Response> {
+
+    private GrpcDgraphResult(final Graphresponse.Response theRootResult) {
+        super(theRootResult);
+    }
+
+    @Override
+    public JsonObject toJsonObject() {
+        final Stack<Graphresponse.Node> nodes = new Stack<>();
+        final Stack<JsonObject> jsonNodes = new Stack<>();
```
Comment (mohitranka) [inline src/main/java/io/dgraph/client/GrpcDgraphResult.java]: @bitmalloc You should use `nodes` to traverse and `nodeToJson` whenever the json representation is needed for the node, without maintaining another stack for `jsonNodes`, which is not really being used.

File: src/test/java/io/dgraph/client/DgraphClientTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
+package io.dgraph.client;
```
Comment (mohitranka) [inline src/test/java/io/dgraph/client/DgraphClientTest.java]: @bitmalloc There should be Dgraph Copyright stub here as well.

File: src/main/java/io/dgraph/client/GrpcDgraphResult.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+        final Graphresponse.Node rootNode = getResponse().getN();
+        final JsonObject rootJson = nodeToJson(rootNode);
+
+        nodes.add(rootNode);
```
Comment (mohitranka) [inline src/main/java/io/dgraph/client/GrpcDgraphResult.java]: @bitmalloc Is there any difference expected between to use `.add` for `nodes` stack and and `.push` for `jsonNodes` stack?

Comment (bitmalloc) [inline src/main/java/io/dgraph/client/GrpcDgraphResult.java]: those methods are identical, but I guess for the context of the stack `push` should be more clear, I can change that.

File: src/main/java/io/dgraph/client/GrpcDgraphResult.java (hunk 1 above)
Comment (bitmalloc) [inline src/main/java/io/dgraph/client/GrpcDgraphResult.java]: yeah, I forgot to remove that extra stack, thanks for the catch.

File: src/test/java/io/dgraph/client/DgraphClientTest.java (hunk 2 above)
Comment (bitmalloc) [inline src/test/java/io/dgraph/client/DgraphClientTest.java]: done.

**Comment by bitmalloc** [comment]
I just pushed a commit with changes from the comments. Thanks.

**Comment by mohitranka** [comment]
LGTM. Assigning it to @manishrjain for review + merge.

**Comment by pawanrawal** [comment]
Thanks @bitmalloc for working on this and @mohitranka for verifying this. @bitmalloc maybe we should remove the `.proto` file from this repo and have a link to the one in the Dgraph repo in `README.md`. 
Apart from that LGTM. You can squash your commits and push to master.

---

Reviewed 16 of 18 files at r1, 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph4j/1#-:-KLehXE1L2g63zsY1aDZ:1979118872)*

**Comment by bitmalloc** [comment]
Great! Thanks for the review guys, I'll squash and merge to master. Closing this PR.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph4j/1#-:-KLhpgpGOmK5Jn6YrSzX:-1839680688)*

**Pull request closed without merging**